gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/kmeans/KMeans.java

```diff
@@ -0,0 +1,147 @@
+ */
+package org.gradoop.flink.model.impl.operators.kmeans;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.functions.KeySelector;
+import org.apache.flink.api.java.operators.IterativeDataSet;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.utils.DataSetUtils;
+import org.gradoop.common.model.api.entities.Edge;
+import org.gradoop.common.model.api.entities.GraphHead;
+import org.gradoop.common.model.api.entities.Vertex;
+import org.gradoop.flink.model.api.epgm.BaseGraph;
+import org.gradoop.flink.model.api.epgm.BaseGraphCollection;
+import org.gradoop.flink.model.api.operators.UnaryBaseGraphToBaseGraphOperator;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.SelectNearestCenter;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.VertexPostProcessingMap;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CountAppender;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CentroidAccumulator;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.CentroidAverager;
+import org.gradoop.flink.model.impl.operators.kmeans.util.Centroid;
+import org.gradoop.flink.model.impl.operators.kmeans.util.Point;
+
+import java.util.Objects;
+
+/**
+ * Takes a logical graph, an user defined amount of iterations and centroids, and the property names
+ * of the vertex that are used for the clustering as input. Adds the clusterId, together with the
+ * cluster coordinates to the properties of the vertex. The datatype of the properties can be any numeric
+ * value. Returns the logical graph with modified vertex properties.
+ *
+ * @param <G>  The graph head type.
+ * @param <V>  The vertex type.
+ * @param <E>  The edge type.
+ * @param <LG> The type of the graph.
+ * @param <GC> The type of the graph collection.
+ */
+public class KMeans<G extends GraphHead, V extends Vertex, E extends Edge,
+  LG extends BaseGraph<G, V, E, LG, GC>, GC extends BaseGraphCollection<G, V, E, LG, GC>> implements
+  UnaryBaseGraphToBaseGraphOperator<LG> {
+
+  /**
+   * Number of iterations
+   */
+  private final int iterations;
+
+  /**
+   * Amount of clusters
+   */
+  private final int centroids;
+
+  /**ng
+   * Name of the first spatial property used for the clustering
+   */
+  private final String firstPropertyName;
+
+  /**
+   * Name of the second spatial property used for the clustering
+   */
+  private final String secondPropertyName;
+
+  /**
+   * Constructor to create an instance of KMeans
+   *
+   * @param iterations        Number of iterations, e.g., 20
+   * @param centroids         Amount of centroids that are determined by the algorithm
+   * @param propertyNameOne   First spatial property name of the vertices
+   * @param propertyNameTwo   Second spatial property name of the vertices
+   */
+  public KMeans(int iterations, int centroids, String propertyNameOne, String propertyNameTwo) {
+    Objects.requireNonNull(propertyNameOne);
```

Comment:
Since this function returns the argument, it can be directly used in the assignment in line 88. Same for the check one line below.

Reply:
Done